### PR TITLE
Add GroundTruth models #30

### LINKED ISSUE #30: Add models for groundtruth_uri and taskdata_uri contents
(see related ticket for details)

See for examples: https://github.com/IntuitionMachines/hmt-servers/issues/3438#issuecomment-592769518

### PR DESCRIPTION
**Pull request by mtratsiuk**
Closes #30

### CODE REVIEW

File: basemodels/__init__.py [hunk 1]
```diff
@@ -1,2 +1,4 @@
-from .manifest import Manifest, NestedManifest, RequestConfig, TaskData, Webhook
+from .manifest import validate_manifest_uris, Manifest, NestedManifest, RequestConfig, TaskData, Webhook
 from .via import ViaDataManifest
+from .groundtruth import validate_groundtruth_entry
```
Comment (gaieges) [inline basemodels/__init__.py:3]: might want to make these part of a `.manifest.data` submodule as its tied inherently to the manifest.

File: basemodels/manifest.py [hunk 2]
```diff
@@ -239,3 +244,44 @@ def validate_taskdata_uri(self, data, value):
     validate_taskdata = validate_taskdata_uri
 
     webhook = ModelType(Webhook)
+
+
+def traverse_json_entries(data: Any, callback: Callable) -> int:
```
Comment (gaieges) [inline basemodels/manifest.py:249]: I presume this is prep for the streaming idea you were working with?  Could use a comment as such, if thats correct.

File: basemodels/taskdata.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        "datapoint_hash": "f4acbe8562907183a484498ba901bfe5c5503aaa"
+      },
+      {
+        "task_key": "20bd4f3e-4518-4602-b67a-1d8dfabcce0c",
+        "datapoint_uri": "https://domain.com/file2.jpg",
+        "datapoint_hash": "f4acbe8562907183a484498ba901bfe5c5503aaa"
+      }
+    ]
+    """
+    task_key = UUIDType()
```
Comment (Villux) [inline basemodels/taskdata.py:22]: Why we want to set a `task_key`? @gaieges do you have an answer for this? I understand why we have that in hcaptcha, but why we need to set it before tasks enter the system?

File: basemodels/taskdata.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    datapoint_uri = URLType(required=True, min_length=10)
+    datapoint_hash = StringType()
```
Comment (Villux) [inline basemodels/taskdata.py:24]: Same for the hash? Is this just to guarantee that the image behind the url is the same? @gaieges

File: basemodels/taskdata.py (hunk 3 above)
Comment (gaieges) [reply basemodels/taskdata.py:22]: That's why we made it optional. We don't need it necessarily at dashboard validation time.  Perhaps down the road we can make a separate model that is meant for post-loader/post-task-processor time

File: basemodels/taskdata.py (hunk 4 above)
Comment (gaieges) [reply basemodels/taskdata.py:24]: Yep same idea, this we don't actually use afaik still

**Pull request merged**